keras_cv/models/segmentation/deeplab.py

```diff
@@ -67,12 +73,21 @@ def __init__(
         backbone_weights=None,
         spatial_pyramid_pooling=None,
         segmentation_head=None,
+        segmentation_head_activation="softmax",
```

Comment:
I think the `segmentation_head` should be responsible for the activation

Reply:
Yup - I don't remember why I separated them, but it seemed sensible at the time... Updating

Reply:
Oh I remember - I wanted to separate them to allow a backbone to use logits instead of a softmax, in which case we could turn off the softmax for the segmentation head.

Updated to have the `output_activation` of the head optional within the semantic segmentation head class (output can be logits or softmaxed probability distribution).